pgjdbc/src/main/java/org/postgresql/jdbc/ArrayDecoding.java

```diff
@@ -336,7 +336,7 @@ Object parseValue(String stringVal, BaseConnection connection) throws SQLExcepti
 
     @Override
     Object parseValue(String stringVal, BaseConnection connection) throws SQLException {
-      return connection.getTimestampUtils().toDate(null, stringVal);
+      return connection.getTimestampUtils().toDate(null, stringVal.getBytes());
```

Comment:
I'm afraid I don't understand how this was supposed to improve the performance.
It seems extremely misleading to pass `byte[]`. Initially I thought it was supposed to be a `postgresql binary encoded value`, however, in practice it the method expects a sting in a format of `byte[]`.

Adding `.getBytes()` all over the place does not make life easier as well :-/

Reply:
Ah, the core of the change is to avoid `byte[]->String->byte[]` when parsing `text-encoded` dates from the backend. However, it looks like the new implementation might affect cases like `getDate()` which falls back to `.getBytes()` extra conversion.